homeassistant/components/climate/maxcube.py

```diff
@@ -85,8 +91,15 @@ def current_temperature(self):
 
     @property
     def current_operation(self):
-        """Return current operation (auto, manual, boost, vacation)."""
+        """Return the current operation."""
         device = self._cubehandle.cube.device_by_rf(self._rf_address)
+        """If we are only interested in whether the TRV is off or on."""
+        if self._trv_on_off and hasattr(device, 'valve_position'):
```

Comment:
With this option activated the device is turned into a binary device, from a 4 mode device. But this isn't reflected in the GUI, which still shows 4 modes selectable. How should this be handled? Should the list of operation modes be changed, with this option activated, to only allow `STATE_ON` and `STATE_OFF`?

I would move all this logic to the `update` method below, and add a private instance attribute `_operation_mode` that should hold the current operation mode and just return that here.